lib/rules/rule-empty-line-before/README.md

```diff
@@ -0,0 +1,244 @@
+# rule-empty-line-before
+
+Require or disallow an empty line before rules.
+
+```css
+a {}
+      /* ← */
+b {}  /* ↑ */
+/**      ↑
+ * This line */
+```
+
+If the rule is the very first node in a stylesheet then it is ignored.
+
+## Options
+
+`string`: `"always"|"never"|"always-multi-line"|"never-multi-line"`
+
+### `"always"`
+
+There *must always* be an empty line before rules.
+
+The following patterns are considered warnings:
+
+```css
+a {} b {}
+```
+
+```css
+a {}
+b {}
+```
+
+The following patterns are *not* considered warnings:
+
+```css
+a {}
+
+b {}
+```
+
+### `"never"`
+
+There *must never* be an empty line before rules.
+
+The following patterns are considered warnings:
+
+```css
+a {}
+
+b {}
+```
+
+The following patterns are *not* considered warnings:
+
+```css
+a {} b {}
+```
+
+```css
+a {}
+b {}
+```
+
+### `"always-multi-line"`
+
+There *must always* be an empty line before multi-line rules.
+
+The following patterns are considered warnings:
+
+```css
+a
+{}
+b
+{}
+```
```

Comment:
I think these are confusing examples. Multi-line rules typically manifest as multiple lines of declarations — that's what people will recognize. Let's do that.